src/jbms/argparse.cpp

```diff
@@ -54,13 +131,791 @@ static bool starts_with(string_view s, string_view prefix) {
   return s.size() >= prefix.size() && memcmp(&s[0], &prefix[0], prefix.size() * sizeof(string_view::value_type)) == 0;
 }
 
-// This converts an ASCII string to uppercase.  This is broken for UTF-8 since in unicode and especially UTF-8 upper case is not a code-unit-wise operation.
-static string ascii_to_upper(string_view s) {
-  string result;
-  result.reserve(s.size());
-  for (auto x : s)
-    result.push_back((char)std::toupper(x));
-  return result;
+static string ascii_to_upper(string s) {
+  /**
+   * \brief Converts UTF-8 encoded string to uppercase.
+   * \param s String to convert to uppercase.
+   * \returns Uppercased string.
+   */
+  size_t len = s.length();
+  if (len == 0)
+    return s;
+  // convert the string into a vector of 'unsigned char' type, this is
+  // neccessary as in the following code we'll be comparing the string elements
+  // against type that exceeds the basic 'char' type
+  vector<__uint8_t> str(s.data(), s.data() + len + 1);
```

Comment:
@jbms `signed char` will not work in this case as the `case` labels are of `unsigned char` and that is exactly what we need to do in-order to even identify a UTF-8 character. I'm using `__uint8_t` type instead of writing `unsigned char`, do you want me to change it to `unsigned char`?